### PR TITLE
Fix TSQL UDF BEGIN parsing

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Addresses some leftover issues in #970

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -877,10 +877,19 @@ def _parse_statement(self) -> t.Optional[exp.Expression]:
             return self.STATEMENT_PARSERS[self._prev.token_type](self)
 
         if self._match_set(Tokenizer.COMMANDS):
+            command_kind = self._prev.text
+            command_body = self._curr.text if self._curr else ""
```
Comment (tobymao) [inline sqlglot/parser.py:881]: why is this part necessary?

Comment (GeorgeSittas) [reply sqlglot/parser.py:881]: In the tokenizer, line 847, we have this condition for tokenizing whatever comes after a command as a string:
```python
        if token_type in self.COMMANDS and (
            len(self.tokens) == 1 or self.tokens[-2].token_type == TokenType.SEMICOLON
        ):
```
This works if we have for example a single command to parse, like `SET foo bla bar`, because it will tokenize it as `<COMMAND, text: SET> <STRING, text: foo bla bar>`. The problem with the UDF was that the above condition would never be true because the command appears in the middle of a statement, so we'd tokenize the above like `<COMMAND, text: SET> <VAR, text: foo> <VAR, text: bla> <VAR, text: bar>`, which is harder to parse. The code I added makes sure we consume all remaining tokens and use their texts to construct a string, like it's done for the simple case in the tokenizer.

Comment (GeorgeSittas) [reply sqlglot/parser.py:881]: The while loop I added should be a no-op for the general case, because `self._curr` will be `None`.

File: sqlglot/tokens.py [hunk 2]
```diff
@@ -844,8 +844,11 @@ def _add(self, token_type: TokenType, text: t.Optional[str] = None) -> None:
         )
         self._comments = []
 
+        # If we have either a semicolon or a begin token before the command's token, we'll parse
+        # whatever follows the command's token as a string
         if token_type in self.COMMANDS and (
-            len(self.tokens) == 1 or self.tokens[-2].token_type == TokenType.SEMICOLON
+            len(self.tokens) == 1
+            or self.tokens[-2].token_type in (TokenType.SEMICOLON, TokenType.BEGIN)
```
Comment (tobymao) [inline sqlglot/tokens.py:851]: make this a constant and a set

**Review by tobymao — APPROVED** [review]

File: sqlglot/tokens.py (hunk 2 above)
Comment (GeorgeSittas) [reply sqlglot/tokens.py]: done
